browser-test/src/application_navigation.test.ts

```diff
@@ -75,59 +77,229 @@ describe('Applicant navigation flow', () => {
+        await applicantQuestions.answerDateQuestion('2021-11-01')
+
+        await applicantQuestions.clickPrevious()
+
+        // Verify we're taken to the previous page (which is the review page
+        // since this was the first block) page and the answers were saved
+        await applicantQuestions.expectReviewPage()
+        await applicantQuestions.expectQuestionAnsweredOnReviewPage(
+          dateQuestionText,
+          '11/01/2021',
+        )
+        await applicantQuestions.expectQuestionAnsweredOnReviewPage(
+          emailQuestionText,
+          'test1@gmail.com',
+        )
+      })
+
+      it('error on previous modal > click previous without saving > answers not saved', async () => {
+        const {page, applicantQuestions} = ctx
+        await loginAsAdmin(page)
+        await enableFeatureFlag(page, 'save_on_all_actions')
+        await logout(page)
+
+        await applicantQuestions.applyProgram(programName)
+        await applicantQuestions.answerDateQuestion('2021-11-01')
+        await applicantQuestions.answerEmailQuestion('')
+
+        await applicantQuestions.clickPrevious()
+        await applicantQuestions.expectErrorOnPreviousModal()
+
+        // Proceed to the previous page (which will be the review page,
+        // since this is the first block), acknowledging that answers won't be saved
+        await applicantQuestions.clickPreviousWithoutSaving()
+
+        await applicantQuestions.expectReviewPage()
+        await applicantQuestions.validateNoPreviouslyAnsweredText(
+          dateQuestionText,
+        )
+        await applicantQuestions.validateNoPreviouslyAnsweredText(
+          emailQuestionText,
+        )
+      })
+
+      it('error on previous modal > click previous without saving > shows previous block', async () => {
+        const {page, applicantQuestions} = ctx
+        await loginAsAdmin(page)
+        await enableFeatureFlag(page, 'save_on_all_actions')
+        await logout(page)
+
+        await applicantQuestions.applyProgram(programName)
+        await applicantQuestions.answerDateQuestion('2021-11-01')
+        await applicantQuestions.answerEmailQuestion('test1@gmail.com')
+        await applicantQuestions.clickNext()
+        // Nothing to fill in since this is the static question block
+        await applicantQuestions.clickNext()
+
+        // Don't fill in the address question, and try going to previous block
+        await applicantQuestions.clickPrevious()
+        await applicantQuestions.expectErrorOnPreviousModal()
+
+        // Proceed to the previous page and verify the first block answers are present
+        await applicantQuestions.clickPreviousWithoutSaving()
+        // This is the static question block, so continue to the previous block
+        await applicantQuestions.clickPrevious()
+
+        await applicantQuestions.checkDateQuestionValue('2021-11-01')
+        await applicantQuestions.checkEmailQuestionValue('test1@gmail.com')
```

Comment:
**question**: should this also be testing that the answers weren't saved, or is that covered by the `error on previous modal > click previous without saving > answers not saved` case?

Reply:
Right, I was thinking the `error on previous modal > click previous without saving > answers not saved` case handled that. Those test cases could be combined if that's preferable, but I thought it was nice to test the two different expected behavior separately.